dev/tools/controllerbuilder/pkg/io/cache.go

```diff
@@ -17,22 +17,38 @@ package io
 import (
 	"context"
 	"fmt"
+	"io"
+	"net/http"
 	"os"
 )
 
 // WriteToCache write `out` to a temporary file under `dir`.
-func WriteToCache(ctx context.Context, dir string, out string, namePattern string) (string, error) {
-	if _, err := os.Stat(dir); os.IsNotExist(err) {
-		if err := os.Mkdir(dir, 0755); err != nil {
-			return "", fmt.Errorf("create dir %s: %w", dir, err)
+func WriteToCache(ctx context.Context, dest string, out string, namePattern string) (string, error) {
+	if dest == "" {
+		return "", fmt.Errorf("dir is required")
+	}
+
+	// if dest is File, then write to file
+	fi, err := os.Stat(dest)
+	if err != nil {
+		return "", err
+	}
+	if fi.Mode().IsRegular() {
+		return dest, os.WriteFile(dest, []byte(out), 0644)
+	}
+
+	// If dest is dir, then write to dir
+	if _, err := os.Stat(dest); os.IsNotExist(err) {
```

Comment:
I think you can just call `os.MkdirAll`, no need to check if dir exists

Reply:
Good catch! Thanks